bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/ghost/GhostAsterism.scala

```diff
@@ -0,0 +1,368 @@
+package edu.gemini.spModel.gemini.ghost
+
+import edu.gemini.spModel.core.{Coordinates, Magnitude, MagnitudeBand, MagnitudeSystem, Offset}
+import edu.gemini.spModel.target.SPTarget
+import edu.gemini.spModel.target.env.Asterism
+
+import java.time.{Duration, Instant}
+
+import scalaz._
+import Scalaz._
+
+/** Base trait for the three GHOST asterism types: two target, beam switching,
+  * and high resolution.
+  */
+sealed trait GhostAsterism extends Asterism {
+  import GhostAsterism.{XBinning, YBinning}
+
+  def xBinning: XBinning
+  def yBinning: YBinning
+}
+
+/** GHOST asterism model.
+  */
+object GhostAsterism {
+
+  /** The GHOST guide fibers can be enabled or disabled for each science target.
+    * Typically they are enabled for bright targets (< B mag 18 for standard
+    * resolution, < B mag 17 for high resolution) but disabled for faint targets.
+    * OIWFS state can also be explicitly disabled, say, in a crowded field where
+    * OIWFS is less effective.
+    */
```

Comment:
This is probably my lack of understanding about the GHOST specifics, but this comment was somewhat confusing to me. It starts talking about the GHOST guide fibers, but then it talks about the OIWFS state - I wonder if we can define first what an OIWFS state is, so it is clear that this state is about the guide fibers? I might be understanding all this wrong, so apologies for the noise in that case.